psi4/driver/driver_nbody.py

```diff
@@ -25,830 +25,1290 @@
+
+        build_nbody_compute_list()
+        --------------------------
+        * initializes dicts for each of nocp, cp, vmfc (2 for this one) with keys requested n-body levels and values empty sets
+        * use combinatorics formulas to fill each key with (frag, bas) indices (what fragments are active and what fragments have basis functions)
+          needed to compute the requested bsse treatments at the requested n-body levels.
+        * merge by n-body level the sets of indices for each bsse treatment into an "all" dict. return this and all the per-bsse dicts.
+
+    * merge all the requested indices, prepend a modelchem treatment index to form `mc_(frag, bas)`
+    * construct a molecule appropriately real/ghosted from active-fragment info in (frag, bas)
+    * if embedding_charges active, prepare external_potentials array for atoms not in bas fragments
+    * for any new `mc_(frag, bas)` index, append a new computer to self.task_list
+
+--------------------------
+ManyBodyComputer.compute()
+--------------------------
+* compute() for each job in self.task_list
+
+----------------------------------
+ManyBodyComputer.get_psi_results()
+----------------------------------
+
+    Computer.get_results()
+    ----------------------
+
+        Computer._prepare_results()
+        ---------------------------
+        * if multiple modelchems (multilevel):
+
+            multilevel.prepare_results()
+            ----------------------------
+            * separates out each modelchem calcs and calls _prepare_results on each
+            * sums modelchem levels and returns small dict back to get_results()
+
+        * call get_results() for each job in task list
+        * assemble all the computed energies, all the computed gradients, and all the computed hessians
+        * for each available derivative, call:
+
+            assemble_nbody_components()
+            ---------------------------
+            * re-call build_nbody_compute_list to get the cp/nocp/vmfc lists again
+
+                build_nbody_compute_list()
+                --------------------------
+
+            * slice up the supersystem mol into fragment atom ranges to help g/h arrays build piecemeal
+            * prepare empty {bsse_type}_by_level and {bsse_type}_body_dict structs. the former have different contents for vmfc
+            * for cp and nocp, resort the build_nbody_compute_list returns into per-body lists suitable for summing
+            * note that nb loops often run over more than active nbodies_per_mc_level item due to 1-body for subtraction and multilevel complications
+            * for each possibly active n-body level and each active bsse_type, call _sum_cluster_ptype_data to build by_level structs
+
+                _sum_cluster_ptype_data()
+                -------------------------
+                * build up ene, grad, Hess in per-fragment pieces based on list of (frag, bas) subjobs active for that bsse treatment
```

Comment:
"build up pieces" is vague.

Reply:
Does this mean "compute ene/grad/Hess" for each fragment using the given n-body formula?